tests/e2e-hello.spec.ts

```diff
@@ -0,0 +1,31 @@
+import { existsSync, mkdirSync, copySync } from "fs-extra";
+import { join } from "path";
+import { GenerateOptions, generate } from "../src/generate";
+
+// End-to-end test function
+async function testCLI() {
+  // Step 1: Create a temp directory
+  const tempDir = join(__dirname, "temp");
+  if (!existsSync(tempDir)) {
+    mkdirSync(tempDir);
+
+    const examplesDir = join(__dirname, "../examples/next/");
+    copySync(examplesDir, tempDir);
+  }
+
+  const options = {
+    projectDir: tempDir,
+    regenerateDescription: true,
+    technology: "next14",
+    model: "gpt-4-1106-preview",
+  } as GenerateOptions;
```

Comment:
better way to do this:
don't use `as`.

but:

```ts
const options: GenerateOptions = {
    projectDir: tempDir,
    regenerateDescription: true,
    technology: "next14",
    model: "gpt-4-1106-preview",
  };
```

This type checks it. `as` doesn't. Explained here:
![CleanShot 2023-11-28 at 23 20 30@2x](https://github.com/alephmatic/aleph0/assets/3090527/a89fd2aa-1f1c-4070-b64b-f3bc08fd2b79)